docs/using-the-jdbc-driver/using-plugins/UsingTheDsqlAuthenticationPlugin.md

```diff
@@ -0,0 +1,37 @@
+# AWS Aurora DSQL IAM Authentication Plugin
+
+## What is IAM?
+AWS Identity and Access Management (IAM) grants users access control across all Amazon Web Services. IAM supports granular permissions, giving you the ability to grant different permissions to different users. For more information on IAM and it's use cases, please refer to the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html).
+
+## Prerequisites
+> [!WARNING]\
+> To use this plugin, you must include the runtime dependency [AWS Java SDK DSQL](https://central.sonatype.com/artifact/software.amazon.awssdk/dsql) in your project.
+>
+> Note: AWS Java SDK DSQL may have transitive dependencies that are also required (ex. [AWS Java SDK Core](https://central.sonatype.com/artifact/software.amazon.awssdk/aws-core/)). If you are not using a package manager such as Maven or Gradle, please refer to Maven Central to determine these transitive dependencies.
+
+To enable the AWS Aurora DSQL IAM Authentication Plugin, add the plugin code `iamDsql` to the [`wrapperPlugins`](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters) value, or to the current [driver profile](../UsingTheJdbcDriver.md#connection-plugin-manager-parameters).
+
+> [!WARNING]\
+> The `iam` plugin must NOT be specified when using the `iamDsql` plugin.
+
+This plugin requires valid AWS credentials. See more details at [AWS Credentials Configuration](../custom-configuration/AwsCredentialsConfiguration.md).
+
+## AWS Aurora DQSL IAM Database Authentication
+The AWS JDBC Driver supports Amazon AWS Identity and Access Management (IAM) authentication. When using AWS IAM database authentication, the host URL must be a valid Amazon endpoint, and not a custom domain or an IP address.
+<br>ie. `cluster-identifier.dsql.us-east-1.on.aws`
```

Comment:
Host URL could be a an IP address or can use a custom domain or CNAME alias. `iamHost` parameter should be used in this case.

Reply:
DSQL doesn't currently support custom domains or IP addresses. It requires an SNI that matches the cluster endpoint during the TLS handshake. If you specify an IP address instead you get an error like this when connecting:

```
Exception in thread "main" org.postgresql.util.PSQLException: FATAL: unable to accept connection, sni was not received
```